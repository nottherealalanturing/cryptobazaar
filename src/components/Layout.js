import React from 'react';
import { Link, Outlet } from 'react-router-dom';
import { FcHome, FcSearch } from 'react-icons/fc';
import styles from './styles/Layout.module.css';

const Layout = () => (
  <div>
    <nav className={styles.navbar}>
      <Link to="/" className="homeNav">
        <FcHome />
      </Link>
      <header className="titleNav">name</header>
      <span className="searchNav">
        <input className="searchInput" />
        <FcSearch className="searchLogo" />
      </span>
    </nav>

    <div className="content">
      <Outlet />
    </div>
  </div>
);

export default Layout;
