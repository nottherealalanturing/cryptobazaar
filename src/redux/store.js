import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import cryptoReducer from './crypto/cryptoSlice';
import searchReducer from './search/searchSlice';

const reducers = {
  crypto: cryptoReducer,
  search: searchReducer,
};

const store = configureStore({
  reducer: reducers,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
