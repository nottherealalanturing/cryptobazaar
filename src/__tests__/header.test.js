import { Provider } from 'react-redux';
import { HashRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Layout from '../components/Layout';
import store from '../redux/store';

it('Home Components Renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Layout />
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
