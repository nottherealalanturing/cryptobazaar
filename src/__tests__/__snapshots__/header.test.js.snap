// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Components Renders correctly 1`] = `
<div>
  <nav
    className="navbar"
  >
    <a
      className="homeNav"
      href="#/"
      onClick={[Function]}
    >
      <svg
        enableBackground="new 0 0 48 48"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        version="1"
        viewBox="0 0 48 48"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polygon
          fill="#E8EAF6"
          points="42,39 6,39 6,23 24,6 42,23"
        />
        <g
          fill="#C5CAE9"
        >
          <polygon
            points="39,21 34,16 34,9 39,9"
          />
          <rect
            height="5"
            width="36"
            x="6"
            y="39"
          />
        </g>
        <polygon
          fill="#B71C1C"
          points="24,4.3 4,22.9 6,25.1 24,8.4 42,25.1 44,22.9"
        />
        <rect
          fill="#D84315"
          height="16"
          width="12"
          x="18"
          y="28"
        />
        <rect
          fill="#01579B"
          height="6"
          width="6"
          x="21"
          y="17"
        />
        <path
          d="M27.5,35.5c-0.3,0-0.5,0.2-0.5,0.5v2c0,0.3,0.2,0.5,0.5,0.5S28,38.3,28,38v-2C28,35.7,27.8,35.5,27.5,35.5z"
          fill="#FF8A65"
        />
      </svg>
    </a>
    <header
      className="titleNav"
    >
      CryptoBazaar
    </header>
    <span
      className="searchNav"
    >
      <input
        className="searchInput"
        id="search"
        onChange={[Function]}
        placeholder="Search Coin"
        type="text"
        value=""
      />
      <svg
        className="searchLogo"
        enableBackground="new 0 0 48 48"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        version="1"
        viewBox="0 0 48 48"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="#616161"
        >
          <rect
            height="17"
            transform="matrix(.707 -.707 .707 .707 -15.154 36.586)"
            width="4"
            x="34.6"
            y="28.1"
          />
          <circle
            cx="20"
            cy="20"
            r="16"
          />
        </g>
        <rect
          fill="#37474F"
          height="12.3"
          transform="matrix(.707 -.707 .707 .707 -15.839 38.239)"
          width="4"
          x="36.2"
          y="32.1"
        />
        <circle
          cx="20"
          cy="20"
          fill="#64B5F6"
          r="13"
        />
        <path
          d="M26.9,14.2c-1.7-2-4.2-3.2-6.9-3.2s-5.2,1.2-6.9,3.2c-0.4,0.4-0.3,1.1,0.1,1.4c0.4,0.4,1.1,0.3,1.4-0.1 C16,13.9,17.9,13,20,13s4,0.9,5.4,2.5c0.2,0.2,0.5,0.4,0.8,0.4c0.2,0,0.5-0.1,0.6-0.2C27.2,15.3,27.2,14.6,26.9,14.2z"
          fill="#BBDEFB"
        />
      </svg>
    </span>
  </nav>
  <div
    className="content"
  />
</div>
`;
